### PR TITLE
Very small perf tweak to help box2

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

Small tweak to `intersperseFn` which is used by Box2 when you ask for gaps.

On my machine it makes box2 with 100 children 10% faster. In practice it probably won't be so dramatic since I don't think we use box2 with a lot of children and gap. When the children count is small it's basically the same thing as the reduce version.

It does have fewer allocations which could help with GC /shrug

### CODE REVIEW

File: shared/util/arrays.js [hunk 1]
```diff
@@ -6,13 +6,13 @@ export function intersperseFn<A, B>(
   arr: Array<A>
 ): Array<A | B> {
   if (arr.length === 0) {
-    return []
+    // $FlowIssue an array with no length is a valid type for any Array
+    return arr
   }
```
Comment (chrisnojima) [inline shared/util/arrays.js]: Could try and make a local module const that's returned (this is fine also. Making a new array each time likely bad)

File: shared/util/arrays.js [hunk 2]
```diff
@@ -6,13 +6,13 @@ export function intersperseFn<A, B>(
...
 
-  return arr.slice(1).reduce(
-    (acc, x, i, a) => {
-      return acc.concat([separatorFn(i, x, a), x])
-    },
-    [arr[0]]
-  )
+  let toReturn = [arr[0]]
```
Comment (chrisnojima) [inline shared/util/arrays.js]: Not that I think we lose a lot of perf here ever but likely faster
toReturn = new Array(arr. length)
For...
toReturn[i] = ...

Comment (MarcoPolo) [reply shared/util/arrays.js]: I forgot you could size hint the array!

**Comment by buoyad** [comment]
@MarcoPolo the `letterSpacing` in storyshots was fixed in master

**Comment by MarcoPolo** [comment]
Proof of correctness: https://runkit.com/marcopolo/5cbfdc169fa32800123da8fe

**Pull request merged**